RecoLocalCalo/HGCalRecProducers/plugins/HGCalCLUEAlgo.icc

```diff
@@ -534,11 +541,15 @@ void HGCalCLUEAlgo::computeThreshold() {
   }
 }
 
-void HGCalCLUEAlgo::setDensity(const unsigned int layerId) {
+template <typename T>
+void HGCalCLUEAlgoT<T>::setDensity(const unsigned int layerId) {
   auto& cellsOnLayer = cells_[layerId];
   unsigned int numberOfCells = cellsOnLayer.detid.size();
   for (unsigned int i = 0; i < numberOfCells; ++i)
     density_[cellsOnLayer.detid[i]] = cellsOnLayer.rho[i];
 }
 
-Density HGCalCLUEAlgo::getDensity() { return density_; }
+template <typename T>
+Density HGCalCLUEAlgoT<T>::getDensity() {
+  return density_;
+}
```

Comment:
```suggestion
}

// definition of explicit template instantiation
template class HGCalCLUEAlgoT<HGCalLayerTiles>;
template class HGCalCLUEAlgoT<HFNoseLayerTiles>;
```

Reply:
does this remove the need for explicit specialization of every method as it was before?
If so, I think that this is a good solution.

Reply:
I think so - it does compile.